pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/filter/NumericalFilterOptimizer.java

```diff
@@ -75,33 +85,41 @@ Expression optimizeChild(Expression filterExpression, @Nullable Schema schema) {
     Function function = filterExpression.getFunctionCall();
     FilterKind kind = FilterKind.valueOf(function.getOperator());
     switch (kind) {
-      case IS_NULL:
-      case IS_NOT_NULL:
-        // No need to try to optimize IS_NULL and IS_NOT_NULL operations on numerical columns.
+      case BETWEEN: {
+        // Verify that value is a numeric column before rewriting.
+        List<Expression> operands = function.getOperands();
+        Expression value = operands.get(0);
```

Comment:
(minor) Suggest naming it `lhs`. `value` is a little bit misleading here as it is usually a column or a function.

Currently for other filter kinds we check whether `rhs` is numeric but not here. That check itself seems unnecessary, and we can consider consolidating the logic by only checking if `lhs` is numeric